electron/app/protocol-manager.ts

```diff
@@ -0,0 +1,135 @@
+  }
+
+export abstract class ProtocolUtils {
+    // public static setDefaultProtocolClient(): void {
+    //     if (!app.isDefaultProtocolClient(appConstants.PROTOCOL_NAME)) {
+    //         // Define custom protocol handler.
+    //         // Deep linking works on packaged versions of the application!
+    //         app.setAsDefaultProtocolClient(appConstants.PROTOCOL_NAME);
+    //     }
+    // }
+
+    /**
+     * @description Create logic (WIN32 and Linux) for open url from protocol
+     */
+    public static setProtocolHandlerWindowsLinux(): void {
+        // Force Single Instance Application
+        // const gotTheLock = app.requestSingleInstanceLock();
+
+        // app.on('second-instance', (e: Electron.Event, argv: string[]) => {
+        //     // Someone tried to run a second instance, we should focus our window.
+        //     if (WindowManager.mainWindow) {
+        //         if (WindowManager.mainWindow.isMinimized()) WindowManager.mainWindow.restore();
+        //         WindowManager.mainWindow.focus();
+        //     } else {
+        //         // Open main windows
+        //         WindowManager.openMainWindow();
+        //     }
+
+        //     app.whenReady().then(() => {
+        //         WindowManager.mainWindow.loadURL(this._getDeepLinkUrl(argv));
+        //     });
+        // });
+
+        // if (gotTheLock) {
+        //     app.whenReady().then(() => {
+        //         // Open main windows
+        //         WindowManager.openMainWindow();
+        //         WindowManager.mainWindow.loadURL(this._getDeepLinkUrl());
+        //     });
+        // } else {
+        //     app.quit();
+        // }
+        app.whenReady().then(() => {
+            // Open main windows
+            console.log('whenReady');
+            WindowManager.openMainWindow();
+            WindowManager.mainWindow.loadURL(this._getDeepLinkUrl());
+        });
+    }
+
+    /**
+     * @description Create logic (OSX) for open url from protocol
+     */
+    public static setProtocolHandlerOSX(): void {
+        // app.on('open-url', (event: Electron.Event, url: string) => {
+        //     event.preventDefault();
+        //     app.whenReady().then(() => {
+        //         if (!isNil(url)) {
+        //             // Open main windows
+        //             MainWindow.openMainWindow();
+        //             MainWindow.mainWindow.loadURL(this._getUrlToLoad(url));
+        //         } else {
+        //             this._logInMainWindow({ s: 'URL is undefined', isError: true });
+        //             throw new Error('URL is undefined');
+        //         }
+        //     });
+        // });
+        app.on('open-url', (_, rawUrl) => {
+            console.log("on.('open-url'):", rawUrl);
+            const url = new URL(rawUrl);
+    
+            if (url.searchParams.has('token')) {
+                settingsService.updateLoginSettings({ token: url.searchParams.get('token') });
+            }
```

Comment:
Make this a callback so that protocol-manager is not just used for login, but can be used for other things.